app.py

```diff
@@ -0,0 +1,143 @@
+
+    def _annotate(self, mmif: Mmif, **parameters) -> Mmif:
+        # see https://sdk.clams.ai/autodoc/clams.app.html#clams.app.ClamsApp._annotate
+        for arg, val in parameters.items():
+            print("Parameter %s=%s" % (arg, val))
+            # as we defined this `error` parameter in the app metadata
+            if arg == 'error' and val is True:
+                raise Exception("Exception - %s" % parameters['error'])
+
+        # Initialize the MMIF object from the string if needed
+        self.mmif = mmif if type(mmif) is Mmif else Mmif(mmif)
+
+        # process the text documents in the documents list
+        for doc in self.mmif.get_documents_by_type(DocumentTypes.TextDocument):
+            idf_file = parameters['idf_file']
+            feature_dict_file = parameters['feature_dict_file']
+            topn = parameters['topn']
+            new_view = self._new_view(parameters, doc.id)
+            # _run_nlp_tool() is the method that does the actual work
+            self._run_nlp_tool(doc, new_view, doc.id, idf_file, feature_dict_file, topn)
+        # return the MMIF object
+        return self.mmif
+
+    def _new_view(self, runtime_config, docid=None):
+        view = self.mmif.new_view()
+        view.metadata.app = self.metadata.identifier
+        # first thing you need to do after creating a new view is "sign" the view
+        # the sign_view() method will record the app's identifier and the timestamp
+        # as well as the user parameter inputs. This is important for reproducibility.
+        self.sign_view(view, runtime_config)
+        # then record what annotations you want to create in this view
+        # view.new_contain(Uri.DOCUMENT, document=docid)
+        view.new_contain(AnnotationTypes.Alignment)
+        return view
+
+    def _run_nlp_tool(self, doc, new_view, full_doc_id, idf_file, feature_dict_file, topn):
+        """Run the NLP tool over the document and add annotations to the view, using the
+        full document identifier (which may include a view identifier) for the document
+        property."""
+        text = doc.text_value
+
+        # load idf values
+        with open(idf_file, 'rb') as f:
+            idf_values = pickle.load(f)
+
+        # load feature dict
+        with open(feature_dict_file, 'rb') as file:
+            feature_dict = pickle.load(file)
+
+        # restore the values
+        cv = CountVectorizer(vocabulary=feature_dict)
+        tfidf_transformer = TfidfTransformer()
+        tfidf_transformer.idf_ = idf_values
+
+        # get the feature names out into a list
+        feature_names = feature_dict
+
+        # get keywords for the document
+        keywords = get_keywords(text, feature_names, topn, tfidf_transformer, cv)
+
+        # create the document to store the keywords
+        keywords_doc = new_view.new_textdocument(text=keywords)
+
+        a = new_view.new_annotation(AnnotationTypes.Alignment, source=full_doc_id, target=keywords_doc.id)
```

Comment:
let's use `long_id` for safety (https://github.com/clamsproject/mmif/issues/228)